src/compiler/checker.ts

```diff
@@ -7054,6 +7055,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
             function createTypeNodesFromResolvedType(resolvedType: ResolvedType): TypeElement[] | undefined {
                 if (checkTruncationLength(context)) {
+                    if (context.flags & NodeBuilderFlags.NoTruncation) {

```

Comment:
Is there a test that shows what this looks like?

Reply:
Now that you mentioned, there's no test for this scenario. I removed this check and non-regressed. I'm not sure how to test it but if it's utterly necessary I can address that another PR.

Reply:
https://app.codecov.io/gh/microsoft/TypeScript/pull/59332?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=microsoft implies that this code is covered _somewhere_ in a test. Maybe add an assert and reverse engineer a specific test from that?

Reply:
After deeper checks, the file `hugeDeclarationOutputGetsTruncatedWithError.types` access this code.

Check line 18, for the last property `zz`. It changes from `"zz": { ...; }` to `zz: {  };`. This is a "type" tests, emit the "elided" comment gets added: `zz: { /*elided*/ };`

Reply:
Why don't we add `elided` in type tests?

Reply:
I was working on it. Added.

Reply:
Just to confirm, you're saying that in `*.types` files we will write `"zz": { ...; }` but in emit we will write `zz: { /*elided*/ };`? The thing I was interested in is whether or not we should do the latter for both, but maybe that's not what you were describing?

Reply:
No, I mean, before this change files in `*.types` files and during emit we write `"zz": { ...; }`.

After this change, `*.types` files and emit write `"zz": { }`, and `.d.ts` files `"zz": { /*elided*/ }` respectively.

See the test `codeFixClassImplementInterfaceNoTruncationProperties.ts` for the emit example. `hugeDeclarationOutputGetsTruncatedWithError.types` shows the `*.types` change.